arviz/data/io_pyjags.py

```diff
@@ -0,0 +1,321 @@
+        if variable_name in variable_names:
+            parameter_dimension, _, _ = chains.shape
+            if parameter_dimension == 1:
+                variable_name_to_samples_map[variable_name] = chains[0, :, :].transpose()
+            else:
+                variable_name_to_samples_map[variable_name] = np.swapaxes(chains, 0, 2)
+
+    return variable_name_to_samples_map
+
+
+def _extract_arviz_dict_from_inference_data(idata,) -> tp.Mapping[str, np.ndarray]:
+    """
+    Extract the samples dictionary from an ArviZ inference data object.
+
+    Extracts a dictionary mapping parameter names to NumPy arrays of samples
+    with shape (number_of_chains, chain_length, parameter_dimension) from an
+    ArviZ inference data object.
+
+    Parameters
+    ----------
+    idata: InferenceData
+
+    Returns
+    -------
+    a dictionary mapping variable names to NumPy arrays with shape
+             (number_of_chains, chain_length, parameter_dimension)
+
+    """
+    variable_name_to_samples_map = {}
+
+    for key, value in idata.posterior.to_dict()["data_vars"].items():
+        variable_name_to_samples_map[key] = np.array(value["data"])
+
+    return variable_name_to_samples_map
+
+
+def _convert_arviz_dict_to_pyjags_dict(
+    samples: tp.Mapping[str, np.ndarray]
+) -> tp.Mapping[str, np.ndarray]:
+    """
+    Convert and ArviZ dictionary to a PyJAGS dictionary.
+
+    Takes a python dictionary of samples in ArviZ format and returns the samples
+    as a dictionary in PyJAGS format.
+
+    Parameters
+    ----------
+    samples: a dictionary mapping variable names to NumPy arrays with shape
+             (number_of_chains, chain_length, parameter_dimension)
+
+    Returns
+    -------
+    a dictionary mapping variable names to NumPy arrays with shape
+             (parameter_dimension, chain_length, number_of_chains)
+
+    """
+    # pyjags returns a dictionary of NumPy arrays with shape
+    #         (parameter_dimension, chain_length, number_of_chains)
+    # but arviz expects samples with shape
+    #         (number_of_chains, chain_length, parameter_dimension)
+
+    variable_name_to_samples_map = {}
+
+    for variable_name, chains in samples.items():
+        if chains.ndim == 2:
+            number_of_chains, chain_length = chains.shape
+            chains = chains.reshape((number_of_chains, chain_length, 1))
+
+        variable_name_to_samples_map[variable_name] = np.swapaxes(chains, 0, 2)
+
+    return variable_name_to_samples_map
+
+
+def from_pyjags(
+    posterior: tp.Optional[tp.Mapping[str, np.ndarray]] = None,
+    prior: tp.Optional[tp.Mapping[str, np.ndarray]] = None,
+    coords=None,
+    dims=None,
+    save_warmup=None,
+    warmup_iterations: int = 0,
+) -> InferenceData:
+    """
+    Convert PyJAGS posterior samples to an ArviZ inference data object.
+
+    Takes a python dictionary of samples that has been generated by the sample
+    method of a model instance and returns an Arviz inference data object.
```

Comment:
```suggestion
    method of a model instance and returns an Arviz inference data object. 
    For a usage example read the
    :doc:`Cookbook section on from_pyjags </notebooks/InferenceDataCookbook>`
```

This will make it easier for users to find example usage.